### PR TITLE
Implement `listDatabases` for SQLite

### LINKED ISSUE #3260: Implement `listDatabases` for SQLite
### What should be done?

SQLite backend and handler should support `listDatabases` command.

### Where?

SQLite backend and handler.

### Definition of Done

- SQLite handler updated;
- integration/compatibility tests un-skipped.

### PR DESCRIPTION
**Pull request by rumyantseva**
# Description

Closes #3260.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [x] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/backends/sqlite/metadata/registry.go [hunk 1]
```diff
@@ -127,6 +127,28 @@ func (r *Registry) DatabaseList(ctx context.Context) []string {
...
+// If database does not exist, 0 is returned.
+func (r *Registry) DatabaseSize(ctx context.Context, dbName string) (int64, error) {
+	defer observability.FuncCall(ctx)()
+
+	db := r.p.GetExisting(ctx, dbName)
+	if db == nil {
+		return 0, nil
+	}
+
+	q := fmt.Sprintf("SELECT page_count * page_size FROM pragma_page_count(), pragma_page_size()")
```
Comment (rumyantseva) [inline internal/backends/sqlite/metadata/registry.go:141]: https://www.sqlite.org/pragma.html#pragma_page_count

File: internal/backends/sqlite/metadata/registry.go [hunk 2]
```diff
@@ -127,6 +127,28 @@ func (r *Registry) DatabaseList(ctx context.Context) []string {
 	return r.p.List(ctx)
 }
 
+// DatabaseSize returns the size of the database in bytes.
+//
+// If database does not exist, 0 is returned.
+func (r *Registry) DatabaseSize(ctx context.Context, dbName string) (int64, error) {
```
Comment (rumyantseva) [inline internal/backends/sqlite/metadata/registry.go:133]: An alternative approach is to make `DatabaseList` return both names and sizes, but the current approach looks clearer to me.

File: internal/backends/sqlite/metadata/registry.go [hunk 3]
```diff
@@ -127,6 +127,27 @@ func (r *Registry) DatabaseList(ctx context.Context) []string {
...
+	defer observability.FuncCall(ctx)()
+
+	db := r.p.GetExisting(ctx, dbName)
+	if db == nil {
+		return 0, nil
+	}
+
+	row := db.QueryRowContext(ctx, "SELECT page_count * page_size FROM pragma_page_count(), pragma_page_size()")
```
Comment (AlekSi) [inline internal/backends/sqlite/metadata/registry.go:141]: Let's use https://www.sqlite.org/dbstat.html instead

Comment (AlekSi) [reply internal/backends/sqlite/metadata/registry.go:141]: See also #3270

**Review by chilagrow — DISMISSED** [review]
:rocket: 

I know `db.Stats()` queries more than just the size, at the moment our focus isn't performance so it's cool.

**Review by AlekSi — COMMENTED** [review]
On pause until we have dbStats/collStats to reduce the amount of conflicts

**Review by chilagrow — DISMISSED** [review]

**Review by rumyantseva — COMMENTED** [review]
LGTM! (I can't approve though)

File: internal/backends/sqlite/backend.go [hunk 4]
```diff
@@ -75,8 +75,30 @@ func (b *backend) ListDatabases(ctx context.Context, params *backends.ListDataba
...
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		stats, err := db.Stats(ctx, new(backends.DatabaseStatsParams))
+		if backends.ErrorCodeIs(err, backends.ErrorCodeDatabaseDoesNotExist) {
+			stats = new(backends.DatabaseStatsResult)
+			err = nil
+		}
+
```
Comment (AlekSi) [inline internal/backends/sqlite/backend.go:90]: `db.Close()` there

File: internal/backends/sqlite/backend.go [hunk 5]
```diff
@@ -75,8 +75,30 @@ func (b *backend) ListDatabases(ctx context.Context, params *backends.ListDataba
...
+		if err != nil {
+			db.Close()
```
Comment (AlekSi) [inline internal/backends/sqlite/backend.go:92]: not there

File: internal/backends/sqlite/backend.go [hunk 6]
```diff
@@ -75,8 +75,30 @@ func (b *backend) ListDatabases(ctx context.Context, params *backends.ListDataba
...
+			return nil, lazyerrors.Error(err)
+		}
+
+		res.Databases[i] = backends.DatabaseInfo{
+			Name: dbName,
+			Size: stats.SizeTotal,
+		}
+
+		db.Close()
```
Comment (AlekSi) [inline internal/backends/sqlite/backend.go:101]: and not there

File: internal/backends/sqlite/backend.go (hunk 4 above)
Comment (AlekSi) [reply internal/backends/sqlite/backend.go:90]: Actually, let's just remove it: #3327

Comment (chilagrow) [reply internal/backends/sqlite/backend.go:90]: Thanks for this :pray:, `dbc.Close()` removed

**Comment by chilagrow** [comment]
CI will pass once https://github.com/FerretDB/FerretDB/pull/3327 is merged

**Review by AlekSi — APPROVED** [review]

**Comment by rumyantseva** [comment]
@chilagrow I think you need to approve yourself :)
I approve, but I don't have a button :)

**Comment by chilagrow** [comment]
> @chilagrow I think you need to approve yourself :) I approve, but I don't have a button :)

Yes :sweat_smile:

**Review by chilagrow — APPROVED** [review]

**Pull request merged**